src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = [];
+  const dateObject = {};
+  const dateValues = date.split(fromFormat.at(-1));
+
+  for (let i = 0; i < dateValues.length; i++) {
+    dateObject[fromFormat[i]] = dateValues[i];
+  }
+
+  if (fromFormat.includes('YYYY')) {
+    if (toFormat.includes('YY')) {
+      dateObject['YY'] = dateObject['YYYY'].slice(2);
+    }
+  } else if (toFormat.includes('YYYY')) {
+    dateObject['YYYY'] =
+      +dateObject['YY'] < 30
+        ? `20${dateObject['YY']}`
+        : `19${dateObject['YY']}`;
```

Comment:
The logic for converting 'YY' to 'YYYY' is incorrect. The expression `+dateObject['YY'] < 30` will always evaluate to a boolean, which is not intended here. You should compare the numeric value of 'YY' to determine the century correctly. Consider using `parseInt(dateObject['YY'], 10)` for comparison.